### PR TITLE
Fish 7479 create advisor tool documentation

### PR DESCRIPTION
**Pull request by breakponchito**
Adding Upgrade Advisor Tool Documentation

### CODE REVIEW

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
Some suggestions and fixes

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 1]
```diff
@@ -0,0 +1,126 @@
+[[upgrade-advisor-tool]]
+= Upgrade Advisor Tool
+
+This is a CLI (Command Line Interface) tool to advise changes added in Jakarta 10 that affect applications based on Jakarta 8 or 9. The idea of the tool is to summarize all changes that affect your application from the different Jakarta Specs with messages on the logs indicating what code was removed or changed for the different specs listed here: https://jakarta.ee/specifications/
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:4]: already done

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+IMPORTANT: Something important to note is that not all the specs add changes between versions, that is why for the current version some specs won’t show any advises for the implementation code.
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:6]: already removed, thanks

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+[[hot-to-use]]
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:8]: fixed

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+== How to use
+
+You can use directly from command line without configuring anything on your local environment. This is resolved by maven central because this tool was uploaded under the following link: https://mvnrepository.com/artifact/fish.payara.advisor/advisor-maven-plugin[Upgrade Advisor Tool on Maven Central].
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:11]: already done

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+|
+
+|Issue message
+|Message to indicate what issue is causing the advisory
+
+|line 4
+|
+
+|How to proceed message
+|Message to indicate what is the recommended way to proceed with the advisory to make the application compatible with Jakarta 10
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:60]: yeah that is good point, I changed a little to explain the structure of each line and then the description of each part

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+|===
+
+IMPORTANT: The log messages for the advisors are (ASCII) colour-coded to make the user more apprehensive about what they need to change.
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:64]: already removed

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+The following are examples of each of the available types of advisors with the different colors in their structure.
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:66]: already done

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+import jakarta.json.bind.annotation.JsonbProperty;
+
+public class JsonBeanTest {
+    
+    @JsonbProperty(nillable = true)
+    private String name;
+}
+----
+
+For Jakarta 10 the property nillable was marked as deprecated, as a modification for Jakarta JSON Binding 3.0, you can check here: https://jakarta.ee/specifications/jsonb/3.0/[Jakarta JSON Binding 3.0]
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:101]: already done

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+Now with the use of the Upgrade Advisor Tool we can get the advise related to this change as follows:
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:103]: already done

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+image::advisor-tool/advisor-json-binding.png[Advisor JSON Binding]
+
+From the message we can see that the recommendation is to start to use the new annotation JsonbNillable. Then we can have the following code by applying the advise from the tool:
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:107]: already done

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+public class JsonBeanTest {
+    
+    @JsonbProperty
+    @JsonbNillable
+    private String name;
+}
+----
+
+As as conclusion this tool will help us to make our old Jakarta 8/9 applications compatible with last changes added in Jakarta 10.
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:124]: already done

File: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (Pandrex247) [reply docs/modules/ROOT/pages/Technical Documentation/Ecosystem/Advisor Tool.adoc:71]: Not sure you did?

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

**Comment by breakponchito** [comment]
this was closed to change the base branch on a different PR, here the new: https://github.com/payara/Payara-Documentation/pull/294

**Pull request closed without merging**